tokio-trace/test_static_max_level_features/main.rs

```diff
@@ -0,0 +1,75 @@
+#[macro_use]
+extern crate log;
+
+use std::sync::{Arc, Mutex};
+use log::{Level, LevelFilter, Log, Record, Metadata};
+
+#[cfg(feature = "std")]
+use log::set_boxed_logger;
+#[cfg(not(feature = "std"))]
+fn set_boxed_logger(logger: Box<Log>) -> Result<(), log::SetLoggerError> {
+    unsafe {
+        log::set_logger(&*Box::into_raw(logger))
+    }
+}
+
+struct State {
+    last_log: Mutex<Option<Level>>,
+}
+
+struct Logger(Arc<State>);
+
+impl Log for Logger {
```

Comment:
Instead of implementing the `Log` trait from `Log`, we'll want to implement [the `Subscriber` trait](https://docs.rs/tokio-trace-core/0.1.0/tokio_trace_core/subscriber/trait.Subscriber.html) from `tokio-trace`.

Our test will only care about the [`enabled`](https://docs.rs/tokio-trace-core/0.1.0/tokio_trace_core/subscriber/trait.Subscriber.html#tymethod.enabled) and [`event`](https://docs.rs/tokio-trace-core/0.1.0/tokio_trace_core/subscriber/trait.Subscriber.html#tymethod.event) methods, which can function similarly to the `enabled` and `log` methods on the `Log` implemented here. 

We will also have to implement `record`, `record_follows_from`, `enter`, and `exit`, but (like the `flush` method on the `Log` implemented here) those can do nothing. 

For the `new_span` method, we will have to return a non-zero span ID, but we can just pick an arbitrary number and always return it (i.e. `span::Id::from_u64(42)`). When `new_span` is called, we should set the value of the `last_level` to be equal to the new span's level, similarly to what we do on `Event`s.